### PR TITLE
feat(console): add downgrade plan confirm modal

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

feat(console): add downgrade plan confirm modal

### Update
- Support config modal size for the confirm modal
- Support custom action button for the confirm modal since not all confirm modals need a cancel button as its secondary action

## Testing

image

image

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [ ] docs

OR

- [x] This PR is not applicable for the checklist

### CODE REVIEW

File: packages/console/src/pages/TenantSettings/Subscription/DowngradePlanConfirmModal/PlanQuotaDiffList/QuotaDiffItem/index.module.scss [hunk 1]
```diff
@@ -0,0 +1,24 @@
+@use '@/scss/underscore' as _;
+
+.item {
+  display: flex;
+  align-items: center;
+  gap: _.unit(2);
+
+  &.hideChangeState > div::before {
+    content: '・';
```
Comment (xiaoyijun) [inline packages/console/src/pages/TenantSettings/Subscription/DowngradePlanConfirmModal/PlanQuotaDiffList/QuotaDiffItem/index.module.scss:9]: We can use `<ul><li>content</li></ul>` here.

File: packages/console/src/ds-components/ConfirmModal/index.tsx [hunk 2]
```diff
@@ -50,7 +55,10 @@ function ConfirmModal({
         title={title}
         footer={
           <>
-            {onCancel && <Button title={cancelButtonText} onClick={onCancel} />}
+            {customActionButton}
+            {!customActionButton && onCancel && (
```
Comment (charIeszhao) [inline packages/console/src/ds-components/ConfirmModal/index.tsx:59]: This logic is painful. And after offline discussion, I think we should request a design change, to not show a "Contact Us" button on the confirm modal

File: packages/console/src/pages/TenantSettings/Subscription/DowngradeConfirmModalContent/PlanQuotaDiffList/QuotaDiffItem/index.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+type Props = {
+  isChangeStateVisible?: boolean;
+  quotaKey: keyof SubscriptionPlanQuota;
+  quotaValue: SubscriptionPlanQuota[keyof SubscriptionPlanQuota];
+};
+
+function QuotaDiffItem({ isChangeStateVisible = false, quotaKey, quotaValue }: Props) {
+  const isUnlimited = quotaValue === null;
+  const isNotCapable = !isUnlimited && !quotaValue;
+  const isLimited = !isUnlimited && !isNotCapable;
```
Comment (charIeszhao) [inline packages/console/src/pages/TenantSettings/Subscription/DowngradeConfirmModalContent/PlanQuotaDiffList/QuotaDiffItem/index.tsx:25]: I'm confused..

File: packages/console/src/pages/TenantSettings/Subscription/DowngradeConfirmModalContent/PlanQuotaDiffList/QuotaDiffItem/index.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+function QuotaDiffItem({ isChangeStateVisible = false, quotaKey, quotaValue }: Props) {
+  const isUnlimited = quotaValue === null;
+  const isNotCapable = quotaValue === 0 || quotaValue === false;
+  const isLimited = Boolean(quotaValue);
+
+  const Icon = isNotCapable ? Failed : DescendArrow;
+
+  return (
+    <li className={classNames(styles.item, isChangeStateVisible && styles.withChangeState)}>
```
Comment (charIeszhao) [inline packages/console/src/pages/TenantSettings/Subscription/DowngradeConfirmModalContent/PlanQuotaDiffList/QuotaDiffItem/index.tsx:30]: Can we change the name to `hasIcon && styles.hasIcon`? This is easier to understand

File: packages/console/src/pages/TenantSettings/Subscription/DowngradeConfirmModalContent/index.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+          {t('subscription.downgrade_modal.description')}
+        </Trans>
+      </div>
+      <div className={styles.content}>
+        <PlanQuotaDiffList planName={currentPlanName} quotaDiff={diff(targetQuota, currentQuota)} />
+        <PlanQuotaDiffList
+          isTarget
+          planName={targetPlanName}
+          quotaDiff={diff(currentQuota, targetQuota)}
+        />
```
Comment (charIeszhao) [inline packages/console/src/pages/TenantSettings/Subscription/DowngradeConfirmModalContent/index.tsx:39]: No need to `diff` twice? Use a `useMemo` to cache the calculation result

**Review by charIeszhao — APPROVED** [review]
Overall looks good

**Pull request merged**